app/services/users/manage_credits.rb

```diff
@@ -0,0 +1,50 @@
+module Users
+  class ManageCredits
+    def self.call(*args)
+      new(*args).call
+    end
+
+    def initialize(user, user_params)
+      @user = user
+      @organization_id = user_params[:organization_id]
+      @add_credits_param = user_params[:add_credits]
+      @remove_credits_param = user_params[:remove_credits]
+      @add_org_credits_param = user_params[:add_org_credits]
+      @remove_org_credits_param = user_params[:remove_org_credits]
+    end
+
+    def call
+      add_credits if add_credits_param
+      remove_credits if remove_credits_param
+      add_org_credits if add_org_credits_param
+      remove_org_credits if remove_org_credits_param
+    end
+
+    private
+
+    attr_reader :user, :add_credits_param, :add_org_credits_param, :remove_org_credits_param,
+                :remove_credits_param, :organization_id
+
+    def add_credits
```

Comment:
Instead of calling this methods with the inline `if` to check if they can run.

What do you think of using `early returns`?

```ruby
def add_credits
  return unless user_params[:add_credits]
  ....
end
```

Reply:
In this case, I don't know if it can return after the first verification. To do this I need more information from the business case.
For example on the older way, it did verify all the methods and didn't return.